File: full_case_study_solns/_d_mass/matlab_old/hw10/massController.m
classdef massController
    % 
    %    This class inherits other controllers in order to organize multiple controllers.
    %
    %----------------------------
    properties
        zCtrl
        limit
    end
    %----------------------------
    methods
        %----------------------------
        function self = massController(P)
            self.zCtrl = PIDControl(P.kp, P.ki, P.kd, P.F_max, P.beta, P.Ts);
            % plant parameters known to controller
        end
        %----------------------------
        function force = u(self, y_r, y)
            % y_r is the referenced input
            % y is the current state
            z_r = y_r;
            z = y(1);
            
            force_tilde = self.zCtrl.PID(z_r, z, false);
            force = self.saturate(force_tilde);
        end
        %----------------------------
        function out = saturate(self,u)
            if abs(u) > self.limit
                u = self.limit*sign(u);
            end
            out = u;
        end        
    end
end

File: full_case_study_solns/_d_mass/python_old/hw10/massController.py
import numpy as np
import massParamHW10 as P
import sys
sys.path.append('..')  # add parent directory
import massParam as P0
from PIDControl import PIDControl


class massController:

    def __init__(self):
        # Instantiates the PD object
        self.zCtrl = PIDControl(P.kp, P.ki, P.kd, P0.F_max, P0.beta, P0.Ts)
        self.limit = P0.F_max

    def u(self, y_r, y):
        # y_r is the referenced input
        # y is the current state
        z_r = y_r[0]
        z = y[0]

        tau_tilde = self.zCtrl.PID(z_r, z, False)
        tau = self.saturate(tau_tilde)
        return [tau]

    def saturate(self, u):
        if abs(u) > self.limit:
            u = self.limit*np.sign(u)
        return u







